esx_service/vmdk_ops_test.py

```diff
@@ -885,22 +878,29 @@ def test_vmdkops_on_tenant_vm(self):
                                          vm_list=[self.vm1_name]) 
         self.assertEqual(None, error_info)
 
-        # run create command, which should succeed since we have DEFAULT privilege set
+        # run create command, which should fail since "default_datastore" is not set for tenant1
         opts={u'fstype': u'ext4'}
         error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol1_name, opts)
-        self.assertEqual(None, error_info)
-
+        self.assertNotEqual(None, error_info)
+    
         # set access privileges, create a volume with 100MB
-        rights = ["create", "mount"]
         volume_maxsize = "500MB"
         volume_totalsize = "1GB"
-
+        
+        # add access privilege for tenant1, after this, default_datastore will be set to
+        # self.datastore_name
         error_info = auth_api._tenant_access_add(name=self.tenant1_name, 
                                                  datastore=self.datastore_name, 
-                                                 rights=rights, 
+                                                 allow_create=True,
+                                                 switch_default_datastore=False, 
                                                  volume_maxsize=volume_maxsize, 
                                                  volume_totalsize=volume_totalsize)
-
+        
+        # run create command again, which should succeed since "default_datastore" is set for tenant1 now
+        opts={u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol1_name, opts)
+        self.assertEqual(None, error_info)
```

Comment:
we also need to check that the volume is created in correct place. 
We also need to check that if we 'ls' volume it's name does not have datastore if it's on default datastore